File: pa6/java/InputReader.java
import java.io.*;
import java.util.*;


public class InputReader{

    public static void main(String[] args){
        Scanner s = new Scanner(System.in);
	int k = s.nextInt();
	int n = s.nextInt();
	double[][] distances = new double[n][n];
	for(int i = 0; i < n; i++){
	    for(int j = 0; j < n; j++){
                distances[i][j] = Double.parseDouble(s.next());
	    }
	}
        Clusterer.cluster_cost(k, distances);

    }

}

File: pa6/python/inputreader.py
import clusterer


def main():
    k = int(input())
    n = int(input())
    distances = [[0 for i in range(n)] for j in range(n)]
    for i in range(n):
        row = input().strip().split()
        for j in range(len(row)):
            distances[i][j] = float(row[j])
    clusterer.cluster_cost(k, distances)


main()
